recipes/views.py

```diff
@@ -45,15 +50,25 @@ def recipe_detail(request, recipe_slug):
 def recipe_create(request):
     if request.method == "POST":
         form = RecipeCreateForm(request.POST)
-        if form.is_valid():
-            rep = Recipe.objects.create(name=form.cleaned_data["name"])
-            rep.complementary.set(form.cleaned_data["complementary"])
-            rep.tags.set(form.cleaned_data["tags"])
-            return redirect("recipe_detail", recipe_slug=rep.slug)
+        form_steps = StepCreateFormSet(request.POST, prefix="step")
+
+        if form.is_valid() and form_steps.is_valid():
+            recipe = form.save()
+
+            steps = form_steps.save(commit=False)
+            for i, step in enumerate(steps):
+                step.recipe = recipe
+                step.order_id = i
```

Comment:
as discussed, not rendering, just for math of ordering, ok, and yes writing fk's on the object like this is common, snippet from v2:

```
        form = NewPostForm(request.POST, user=request.user, bite_id=bite_id)
        if form.is_valid():
            post = form.save(commit=False)
            post.bite = bite
            post.user = request.user
            submission = form.cleaned_data.get("selected_submission", "").strip()
            if submission:
                post.text = f"{form.cleaned_data['text']}\n\n```{form.cleaned_data['selected_submission']}```"
            post.save()
```